File: anagrams.py
#! /usr/bin/python

import sys

def completeAnagrams(start, end):
	if len(end) >= 1:
		for i in range(1, len(end)):
			#print(start)
			completeAnagrams(start + end[i-1], end[0:i-1] + end[i:len(end)])
		completeAnagrams(start + end[len(end)-1], end[0:len(end) - 1])

completeAnagrams("", sys.argv[1])

File: anagrams.jl
#! /usr/bin/julia

function completeAnagrams(start, ending)
    if length(ending) > 0
        for i in 2:length(ending)
			completeAnagrams(start * ending[i-1], ending[1:i-2] * ending[i:end])
        end
		completeAnagrams(start * ending[end], ending[1:end-1])
    #map(x -> completeAnagrams(start * x, replace(ending, x => "", count=1)), collect(ending))
    else
        #println(start)
    end
end
completeAnagrams("", ARGS[1])
